mupen64plus-core/upstream/src/main/main.c

```diff
@@ -1804,6 +1857,25 @@ m64p_error main_run(void)
     StateChanged(M64CORE_EMU_STATE, M64EMU_RUNNING);
 
     poweron_device(&g_dev);
+
+    DebugMessage(M64MSG_STATUS,"Rom header name = %s",ROM_PARAMS.headername);
+    // PS2 & PS1 needs additional frame count when resetting from in game settings (can use this
+    // count as default if more games get black screen when restarting from in game settings)
+    if( get_r4300_emumode(&g_dev.r4300) >= 2 && strncmp(ROM_PARAMS.headername,"POKEMON STADIUM",15) == 0 )
```

Comment:
This code probably belongs in the front end. Also, if possible, I would try to find a generic solution for all games instead of game this which has a lot of changes of breaking or not working for games that were not tested.